src/csharp/Microsoft.Spark.UnitTest/ConfigurationServiceTests.cs

```diff
@@ -0,0 +1,184 @@
+            }
+
+            {
+                var configService = new ConfigurationService();
+                Environment.SetEnvironmentVariable(
+                    _workerDirEnvVars.WorkerMajorMinorDir.Name, "workerMajorMinorDir");
+
+                Assert.False(_workerDirEnvVars.WorkerMajorMinorBuildDir.IsSet());
+                Assert.True(_workerDirEnvVars.WorkerMajorMinorDir.IsSet());
+                Assert.True(_workerDirEnvVars.WorkerMajorDir.IsSet());
+                Assert.False(_workerDirEnvVars.WorkerDir.IsSet());
+                Assert.Equal(_workerDirEnvVars.WorkerMajorMinorDir.Name, configService.WorkerDirEnvVarName);
+            }
+
+            {
+                var configService = new ConfigurationService();
+                Environment.SetEnvironmentVariable(
+                    _workerDirEnvVars.WorkerMajorMinorBuildDir.Name, "workerMajorMinorBuildDir");
+
+                Assert.True(_workerDirEnvVars.WorkerMajorMinorBuildDir.IsSet());
+                Assert.True(_workerDirEnvVars.WorkerMajorMinorDir.IsSet());
+                Assert.True(_workerDirEnvVars.WorkerMajorDir.IsSet());
+                Assert.False(_workerDirEnvVars.WorkerDir.IsSet());
+                Assert.Equal(
+                    _workerDirEnvVars.WorkerMajorMinorBuildDir.Name, configService.WorkerDirEnvVarName);
+            }
+        }
+
+        [Fact]
+        public void TestWorkerExePathWithNoEnvVars()
+        {
+            var configService = new ConfigurationService();
+
+            Assert.False(_workerDirEnvVars.WorkerMajorMinorBuildDir.IsSet());
+            Assert.False(_workerDirEnvVars.WorkerMajorMinorDir.IsSet());
+            Assert.False(_workerDirEnvVars.WorkerMajorDir.IsSet());
+            Assert.False(_workerDirEnvVars.WorkerDir.IsSet());
+
+            // Environment variables not set, only Microsoft.Spark.Worker filename should be returned.
+            Assert.Equal(ConfigurationService.ProcFileName, configService.GetWorkerExePath());
+        }
+
+        [Fact]
+        public void TestWorkerExePathWithWorkerDirEnvVar()
+        {
+            var configService = new ConfigurationService();
+            var workerDir = "workerDir";
+            Environment.SetEnvironmentVariable(_workerDirEnvVars.WorkerDir.Name, workerDir);
+
+            Assert.False(_workerDirEnvVars.WorkerMajorMinorBuildDir.IsSet());
+            Assert.False(_workerDirEnvVars.WorkerMajorMinorDir.IsSet());
+            Assert.False(_workerDirEnvVars.WorkerMajorDir.IsSet());
+            Assert.True(_workerDirEnvVars.WorkerDir.IsSet());
+
+            // Only WorkerDir is set, WorkerExePath will be built using it.
+            Assert.Equal(
+                Path.Combine(workerDir, ConfigurationService.ProcFileName),
+                configService.GetWorkerExePath());
+        }
+
+        [Fact]
+        public void TestWorkerExePathWithEnvVarPrecedence()
```

Comment:
This could have been just tested with `TestWorkerEnvName`, no? Any particular reason this is tied to `GetWorkerExePath`?

Reply:
Sure I can combine the tests and just test `GetWorkerExePath` instead.